components/ai_chat/core/browser/engine/extended_content_block.h

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_ENGINE_EXTENDED_CONTENT_BLOCK_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_ENGINE_EXTENDED_CONTENT_BLOCK_H_
+
+#include <string>
+#include <variant>
+
+#include "base/values.h"
+
+namespace ai_chat {
+
+enum class ExtendedContentBlockType {
+  // Standard OpenAI content block types
+  kText,
+  kImage,
+  // Customized Brave content block types
+  kPageExcerpt,
+  kChangeTone,
+  kParaphrase,
+  kImprove,
+  kShorten,
+  kExpand,
+};
+
+struct TextContent {
+  std::string text;
+};
+
+struct ImageUrl {
+  ImageUrl();
+  ImageUrl(const ImageUrl&) = delete;
+  ImageUrl& operator=(const ImageUrl&) = delete;
+  ImageUrl(ImageUrl&&);
+  ImageUrl& operator=(ImageUrl&&);
+  ~ImageUrl();
+
+  // Either a URL of the image or the base64 encoded image data.
+  std::string url;
```

Comment:
are we going to validate the url? If so, then we should use `std::variant<GURL, std::string>`

Reply:
Could we put a pin on this and discuss again when I migrate the upload_files and keep it as is for now?